src/pybind/mgr/snap_schedule/fs/schedule.py

```diff
@@ -364,9 +364,9 @@ def add_retention(cls,
             current_retention = json.loads(current)
             for r, v in retention.items():
                 if r in current_retention:
-                    msg = (f'Retention for {r} is already present with value'
-                           f'{current_retention[r]}. Please remove first')
-                    raise ValueError(msg)
+                    msg = (f'Retention for {r} is already present with value '
+                           f'{current_retention[r]}. Please remove it first.')
+                    raise FileExistsError(msg)
```

Comment:
Would this need a QA change anywhere?

Reply:
@mchangir ping?

Reply:
> Would this need a QA change anywhere?

yeah, adding a test case would be apt
I'll get that done

Reply:
test case added